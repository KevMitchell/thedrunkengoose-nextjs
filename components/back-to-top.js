export function BackToTop () {
  const onClick = () => {
    document.body.scrollTop = 0
    document.documentElement.scrollTop = 0
  }

  return (
    <>
      <div className='button' onClick={onClick}/>

      <style jsx>{`
        div {
          position: fixed;
          bottom: 10px;
          right: 10px;
          width: 50px;
          height: 50px;
          background: url('./img/back-to-top.svg');
          background-repeat: no-repeat;
          background-size: contain;
          background-position: center;
          cursor: pointer;
        }
      `}</style>
    </>
  )
}
