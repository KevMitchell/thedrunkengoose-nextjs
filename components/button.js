import Link from 'next/link'
import PropTypes from "prop-types"

export default function Button({ route, label }) {
  return (
    <>
      <Link href={route}>
        <a>{label}</a>
      </Link>

      <style jsx>{`
        div {
          display: block;
          width: 150px;
          height: 40px;
          background-image: url(/img/smear.png);
          background-repeat: no-repeat;
          background-size: contain;
          text-align: center;
          padding-top: 14px;
          font-size: 20px;
          text-decoration: none;
          cursor: pointer;
          color: black;
        }
      `}</style>
    </>
  )
}

Button.propTypes = {
  route: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired
}