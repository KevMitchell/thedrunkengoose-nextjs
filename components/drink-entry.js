import { PropTypes } from 'prop-types'

export function DrinkEntry ({ name, image, ingredients, showInstructions = () => {} }) {
  const composeIngredientList = () => {
    return ingredients.map(ing => {
      const alternativeText = ing.alternative ? ` or ${ing.alternative}` : ''
      const optionalText = ing.optional ? ' (optional)' : ''

      return (
        <div key={name + ing.name}>
          {ing.name + alternativeText + optionalText}
        </div>
      )
    })
  }

  return (
    <div
      className="drinkItem"
      key={name}
      onClick={showInstructions && showInstructions.bind(this, name)}
    >
      <div className="name">
        {name}
      </div>
      {image && (
        <div className="imageContainer">
          <img
            className="image"
            src={`./img/drinks/${image}`}
            alt={name}
          />
        </div>
      )}
      <div className="ingredients">
        {composeIngredientList()}
      </div>
      <style jsx>{`
        .drinkItem {
          display: inline-block;
          background-size: contain;
          background-repeat: no-repeat;
          vertical-align: top;
          width: 300px;
          min-height: 375px;
          padding: 12px;
          margin: 10px;
          cursor: pointer;
          background-image: url('./img/scroll.png')
        }
        
        .image {
          height: 120px;
          margin-top: 6px;
          border-radius: 5px;
        }
        
        .ingredients {
          text-align: left;
          margin-left: 40px;
        }
        
        .drinkRow__substitution {
          font-size: 80%;
          color: rgb(80, 26, 26);
        }
        
        .name {
          font-size: 120%;
          color: rgb(80, 26, 26);
          font-weight: bold;
          margin: 14px 0 10px -8px;
        }
    `}</style>
    </div>
  )
}

DrinkEntry.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  ingredients: PropTypes.arrayOf(PropTypes.object).isRequired,
  showInstructions: PropTypes.func
}
