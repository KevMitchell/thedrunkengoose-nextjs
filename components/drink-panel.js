import { useState, useEffect } from 'react'
import PropTypes from 'prop-types'
import { joinJsx } from '../utils/array-join-of-length'
import { toggleSaved, getSaved } from '../utils/saved-drinks'

export function DrinkPanel ({ name, image, difficulty, base, flavour, ingredients, instructions }) {
  const [saved, setSaved] = useState(false)
  const [expanded, setExpanded] = useState(false)

  useEffect(() => {
    setSaved(getSaved(name))
  }, [])

  const getIngredientText = (ing) => {
    const alternativeText = ing.alternative ? ` or ${ing.alternative}` : ''
    const measure = ing.name.toLowerCase() === 'cherries'
      ? ' '
      : ing.dashes ? ' dashes ' : 'ml '

    const quantity = ing.quantity ? ing.quantity + measure : ''
    let text = quantity + ing.name
    if (ing.optional) text += ' (optional)'
    return (text + alternativeText).toUpperCase().replace(/ML/g, 'ml')
  }

  const toggleSavedState = () => {
    toggleSaved(name, !saved)
    setSaved(!saved)
  }

  return (
    <div
      className={`drinkPanel ${expanded ? 'expanded' : ''}`}
      data-testid='drinkPanel'
    >
      <div className='title'>{name}</div>
      <div className='topPanel'>
        <div className='overview'>
          {image && (
            <div className='imageContainer'>
              <img
                className='image'
                src={`./img/drinks/${image}`}
                alt={name}
              />
            </div>
          )}
          <div className='overviewText'>
            <div className='difficulty'>
              {difficulty}
            </div>
            <div className='base'>
              {base}
            </div>
            <div className='flavour'>
              {flavour.join(', ')}
            </div>
          </div>
        </div>

        <div className='ingredients'>
          {ingredients && joinJsx(ingredients.map(getIngredientText), 44, '  •  ')}
        </div>

        <div className='methodExpander' onClick={() => setExpanded(!expanded)}>
          Method {expanded ? '▲' : '▼'}
        </div>

        <div className='addToSaved' onClick={() => toggleSavedState()}>
          {saved ? '♥' : '♡'} Add to saved
        </div>
      </div>

      <div data-testid='instructions' className={`instructions ${expanded ? 'expanded' : ''}`}>
        {instructions.map((inst, index) =>
          <div
            key={`${name}-${index}`}
            className='instructionLine'>
            {inst}
          </div>
        )}
      </div>

      <style jsx>{`
        .drinkPanel {
          display: block;
          background-color: aliceblue;
          margin: 0 5% 20px 5%;
          height: 320px;
          box-sizing: border-box;
          width: 90%;
          top: 10px;
          z-index: -1;
          -webkit-transition: height 0.5s ease-out;
          -moz-transition: height 0.5s ease-out;
          -o-transition: height 0.5s ease-out;
          -ms-transition: height 0.5s ease-out;
          transition: height 0.5s ease-out;
        }

        .drinkPanel.expanded {
          min-height: 320px;
          height: 520px;
          overflow: visible;
          -webkit-transition: height 0.5s ease-out;
          -moz-transition: height 0.5s ease-out;
          -o-transition: height 0.5s ease-out;
          -ms-transition: height 0.5s ease-out;
          transition: height 0.5s ease-out;
        }

        .topPanel {
          height: 270px;
          position: relative;
        }
        
        .title {
          vertical-align: top;
          padding: 10px;
          border-bottom: 1px solid black;
        }

        .overview {
          padding: 10px;
        }
        
        .imageContainer {
          display: inline-block;
          margin-right: 20px;
        }

        .image {
          height: 120px;
        }

        .overviewText {
          display: inline-block;
          vertical-align: top;
        }

        .difficulty {
          text-align: left;
          padding: 10px 0;
        }

        .base {
          text-align: left;
          padding-bottom: 10px;
        }

        .flavour {
          text-align: left;
          padding-bottom: 10px;
        }

        .methodExpander {
          position: absolute;
          bottom: 0;
          cursor: pointer;
          padding: 5px;
        }

        .addToSaved {
          position: absolute;
          bottom: 0;
          right: 0;
          cursor: pointer;
          padding: 5px;
        }
        
        .title {
          color: grey;
          font-size: 28px;
          text-transform: uppercase;
        }

        .ingredients {
          width: 80%;
          margin: auto;
          letter-spacing: 1px;
        }
        
        .instructions {
          position: relative;
          padding: 20px;
          max-width: 700px;
          text-align: left;
          opacity: 0;
          -webkit-transition: opacity 0.2s ease-out;
          -moz-transition: opacity 0.2s ease-out;
          -o-transition: opacity 0.2s ease-out;
          -ms-transition: opacity 0.2s ease-out;
          transition: opacity 0.2s ease-out;
        }

        .instructions.expanded {
          opacity: 1;
          -webkit-transition: opacity 0.5s ease-out;
          -moz-transition: opacity 0.5s ease-out;
          -o-transition: opacity 0.5s ease-out;
          -ms-transition: opacity 0.5s ease-out;
          transition: opacity 0.5s ease-out;
          transition-delay: 300ms;
        }
        
        .instructionLine {
          display: list-item;
        }
      `}</style>
    </div>
  )
}

DrinkPanel.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  difficulty: PropTypes.string.isRequired,
  base: PropTypes.string.isRequired,
  flavour: PropTypes.arrayOf(PropTypes.string).isRequired,
  ingredients: PropTypes.arrayOf(PropTypes.object).isRequired,
  instructions: PropTypes.arrayOf(PropTypes.string).isRequired
}
