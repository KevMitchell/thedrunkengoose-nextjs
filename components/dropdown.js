import PropTypes from 'prop-types'

export function Dropdown ({ options, id, handleSelect }) {
  return (
    <>
      <div className='dropdown-container'>
        <select
          data-testid={id}
          onChange={handleSelect}
        >
          {options.map(({ name, value, disabled = false }, index) => (
            <option key={index} value={value} disabled={disabled} >
              {name}
            </option>
          ))}
        </select>
        <span className='arrow' tabIndex={0} />
      </div>

      <style jsx>{`
        .dropdown-container {
          position: relative;
          display: inline-block;
          width: 100px;
          height: 50px;
          border: 1px solid #cab488;
        }
        
        select {
          width: 100%;
          height: 100%;
          background: rgba(0, 0, 0, 0);
          color: #cab488;
          border: 0;
          text-align: center;
          text-align-last: center;
          appearance: none;
          -webkit-appearance: none;
          -moz-appearance: none;
          padding: 5px;
        }
        
        .option {
          display: block;
        }
        
        .arrow {
          position: absolute;
          bottom: 0;
          left: 50%;
          display: block;
          height: 10px;
          pointer-events: none;
        }

        .arrow::after {
          --size: 0.35em;
          content: '';
          position: absolute;
          width: 0;
          height: 0;
          
          left: 50%;
          transform: translate(-50%, -50%);

          border-left: var(--size) solid transparent;
          border-right: var(--size) solid transparent;
          border-top: var(--size) solid rgba(255, 255, 255, 0.5);
          top: 45%;
        }
      `}</style>
    </>
  )
}

Dropdown.propTypes = {
  options: PropTypes.array.isRequired,
  id: PropTypes.string,
  handleSelect: PropTypes.func
}
