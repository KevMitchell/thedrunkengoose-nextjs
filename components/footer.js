export default function Footer () {
  return (
    <>
      <footer>
        Copyright Â© Kevin Mitchell, 2021
      </footer>
      <style jsx>{`
        footer {
          position: fixed;
          bottom: 0;
          left: 0;
          opacity: 0.6;
          font-size: 10px;
          margin: 10px;
          color: white;
        }
      `}</style>
    </>
  )
}
