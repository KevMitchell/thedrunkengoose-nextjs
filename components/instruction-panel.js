import PropTypes from 'prop-types'

export function InstructionPanel ({
  name,
  image,
  ingredients,
  instructions,
  closeInstructions,
  visible
}) {
  const getIngredientText = (ing) => {
    const alternativeText = ing.alternative ? ` or ${ing.alternative}` : ''
    const measure = ing.name.toLowerCase() === 'cherries'
      ? ' '
      : ing.dashes ? ' dashes ' : 'ml '

    const quantity = ing.quantity ? ing.quantity + measure : ''
    let text = quantity + ing.name
    if (ing.optional) text += ' (optional)'
    return <div key={text}>{text + alternativeText}</div>
  }

  return (
    <div className={visible ? 'instructionPanel visible' : 'instructionPanel'}>
      <div className='title'>
        Instructions
      </div>
      <div
        className='close'
        onClick={closeInstructions.bind(this)}
      >
        X
      </div>
      <div className='titleBar'>
        {image && (
          <div className='imageContainer'>
            <img
              className='image'
              src={`img/drinks/${image}`}
              alt={name}
            />
          </div>
        )}
        <div className='titleBarText'>
          <div className='name'>
            {name}
          </div>
          <div className='ingredients'>
            {ingredients && ingredients.map(ing => getIngredientText(ing))}
          </div>
        </div>
      </div>
      <div className='instructions'>
        {instructions.map(i =>
          <div
            key={i}
            className='instructionLine'>
            {i}
          </div>
        )}
      </div>

      <style jsx>{`
        .instructionPanel {
          display: block;
          position: fixed;
          background-color: aliceblue;
          margin: 0 5% 0 4%;
          min-height: 400px;
          padding: 15px 20px;
          box-sizing: border-box;
          width: 90%;
          max-width: 600px;
          top: 10px;
          opacity: 0;
          z-index: -1;
          -webkit-transition: opacity 0.5s ease-out;
          -moz-transition: opacity 0.5s ease-out;
          -o-transition: opacity 0.5s ease-out;
          -ms-transition: opacity 0.5s ease-out;
          transition: opacity 0.5s ease-out;
        }
        
        .instructionPanel.visible {
          opacity: 1;
          z-index: 3;
          -webkit-transition: opacity 0.5s ease-out;
          -moz-transition: opacity 0.5s ease-out;
          -o-transition: opacity 0.5s ease-out;
          -ms-transition: opacity 0.5s ease-out;
          transition: opacity 0.5s ease-out;
        }
        
        .titleBar {
          vertical-align: top;
          margin-top: 10px;
        }
        
        .titleBarText {
          display: inline-block;
          vertical-align: top;
        }
        
        .imageContainer {
          display: inline-block;
          margin-right: 20px;
        }
        
        .image {
          height: 120px;
        }
        
        .title {
          color: grey;
          font-style: italic;
          font-size: 24px;
        }
        
        .close {
          position: absolute;
          right: 10px;
          top: 10px;
          color: grey;
          cursor: pointer;
        }
        
        .name {
          vertical-align: top;
          text-align: left;
          font-size: 120%;
          color: darkred;
          font-weight: bold;
          margin-bottom: 20px;
        }
        
        .ingredients {
          text-align: left;
        }
        
        .instructions {
          margin: 20px auto;
          max-width: 700px;
          text-align: left;
        }
        
        .instructionLine {
          display: list-item;
        }
      `}</style>
    </div>
  )
}

InstructionPanel.propTypes = {
  name: PropTypes.string,
  image: PropTypes.string,
  ingredients: PropTypes.array,
  instructions: PropTypes.array,
  closeInstructions: PropTypes.func,
  visible: PropTypes.bool
}
