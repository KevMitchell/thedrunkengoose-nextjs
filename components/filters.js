import PropTypes from 'prop-types'
import { Dropdown } from './dropdown'
import drinks from '../public/data/drinks.json'

const filterUnique = (value, index, self) => self.indexOf(value) === index

export function Filters ({
  handleTypeChange,
  handleDifficultyChange,
  handleBaseChange,
  handleFlavourChange
}) {
  const drinkTypes = {
    options: [
      { name: 'All types', value: 'all' },
      { name: 'Alcoholic', value: 'alcoholic' },
      { name: 'Virgin', value: 'virgin' }
    ],
    handleSelect: handleTypeChange
  }

  const difficultyOptions = drinks
    .map(drink => drink.difficulty)
    .filter(filterUnique)
    .map(opt => ({ name: opt, value: opt }))
  const difficulties = {
    options: [
      { name: 'Any effort', value: 'all' },
      ...difficultyOptions
    ],
    handleSelect: handleDifficultyChange
  }

  const baseOptions = drinks
    .map(drink => drink.base)
    .filter(filterUnique)
    .sort()
    .map(opt => ({ name: opt, value: opt }))
  const bases = {
    options: [
      { name: 'All bases', value: 'all' },
      ...baseOptions
    ],
    handleSelect: handleBaseChange
  }

  const flavourOptions = drinks
    .map(drink => drink.flavour)
    .flat()
    .filter(filterUnique)
    .sort()
    .map(opt => ({ name: opt, value: opt }))
  const flavours = {
    options: [
      { name: 'All flavour', value: 'all' },
      ...flavourOptions
    ],
    handleSelect: handleFlavourChange
  }

  return (
    <div>
      <Dropdown {...drinkTypes} />
      <Dropdown {...difficulties} />
      <Dropdown {...bases} />
      <Dropdown {...flavours} />
      <style jsx>{`
        div {
          margin 20px auto;
        }

        select {
          appearance: none;
          -webkit-appearance: none;
          -moz-appearance: none;
        }

        .arrowClass {
          top: initial;
          bottom: 0;
          left: 50%;
          transform: translate(-50%, -50%);
        }
      `}</style>
    </div>
  )
}

Filters.propTypes = {
  handleTypeChange: PropTypes.func,
  handleDifficultyChange: PropTypes.func,
  handleBaseChange: PropTypes.func,
  handleFlavourChange: PropTypes.func
}
