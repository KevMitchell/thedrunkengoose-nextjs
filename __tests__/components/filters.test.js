import { render } from '@testing-library/react'
import { Filters } from '../../components/filters'

test('renders correct filters', () => {
  const { container } = render(
    <Filters
      handleTypeChange={jest.fn()}
      handleDifficultyChange={jest.fn()}
      handleBaseChange={jest.fn()}
      handleFlavourChange={jest.fn()}
      handleSearchChange={jest.fn()}
    />
  )
  expect(container).toMatchSnapshot()
})
