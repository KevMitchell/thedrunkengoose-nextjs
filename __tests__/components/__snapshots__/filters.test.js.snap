// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct filters 1`] = `
<div>
  <div
    class="jsx-1150314341"
  >
    <div
      class="jsx-1717351303 search-box"
    >
      <div
        class="jsx-1717351303 search-container "
      >
        <div
          class="jsx-1717351303 search-icon"
        />
        <input
          class="jsx-1717351303 search-input"
          title="search-input"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="jsx-1813624227 dropdown-container"
    >
      <select
        class="jsx-1813624227"
        data-testid="type-filter"
      >
        <option
          class="jsx-1813624227"
          value="all"
        >
          All types
        </option>
        <option
          class="jsx-1813624227"
          value="alcoholic"
        >
          Alcoholic
        </option>
        <option
          class="jsx-1813624227"
          value="virgin"
        >
          Virgin
        </option>
      </select>
      <span
        class="jsx-1813624227 arrow"
        tabindex="0"
      />
    </div>
    <div
      class="jsx-1813624227 dropdown-container"
    >
      <select
        class="jsx-1813624227"
        data-testid="difficulty-filter"
      >
        <option
          class="jsx-1813624227"
          value="all"
        >
          Any effort
        </option>
        <option
          class="jsx-1813624227"
          value="Easy"
        >
          Easy
        </option>
        <option
          class="jsx-1813624227"
          value="Medium"
        >
          Medium
        </option>
        <option
          class="jsx-1813624227"
          value="Hard"
        >
          Hard
        </option>
      </select>
      <span
        class="jsx-1813624227 arrow"
        tabindex="0"
      />
    </div>
    <div
      class="jsx-1813624227 dropdown-container"
    >
      <select
        class="jsx-1813624227"
        data-testid="base-filter"
      >
        <option
          class="jsx-1813624227"
          value="all"
        >
          All bases
        </option>
        <option
          class="jsx-1813624227"
          value="Baileys"
        >
          Baileys
        </option>
        <option
          class="jsx-1813624227"
          value="Brandy"
        >
          Brandy
        </option>
        <option
          class="jsx-1813624227"
          value="Cherry brandy"
        >
          Cherry brandy
        </option>
        <option
          class="jsx-1813624227"
          value="Coconut liqueur"
        >
          Coconut liqueur
        </option>
        <option
          class="jsx-1813624227"
          value="Cognac"
        >
          Cognac
        </option>
        <option
          class="jsx-1813624227"
          value="Creme de menthe"
        >
          Creme de menthe
        </option>
        <option
          class="jsx-1813624227"
          value="Elderflower"
        >
          Elderflower
        </option>
        <option
          class="jsx-1813624227"
          value="Galliano"
        >
          Galliano
        </option>
        <option
          class="jsx-1813624227"
          value="Gin"
        >
          Gin
        </option>
        <option
          class="jsx-1813624227"
          value="Ginger beer"
        >
          Ginger beer
        </option>
        <option
          class="jsx-1813624227"
          value="Iced tea"
        >
          Iced tea
        </option>
        <option
          class="jsx-1813624227"
          value="Midori"
        >
          Midori
        </option>
        <option
          class="jsx-1813624227"
          value="Rum"
        >
          Rum
        </option>
        <option
          class="jsx-1813624227"
          value="Tea"
        >
          Tea
        </option>
        <option
          class="jsx-1813624227"
          value="Tequila"
        >
          Tequila
        </option>
        <option
          class="jsx-1813624227"
          value="Vodka"
        >
          Vodka
        </option>
        <option
          class="jsx-1813624227"
          value="Whiskey"
        >
          Whiskey
        </option>
      </select>
      <span
        class="jsx-1813624227 arrow"
        tabindex="0"
      />
    </div>
    <div
      class="jsx-1813624227 dropdown-container"
    >
      <select
        class="jsx-1813624227"
        data-testid="flavour-filter"
      >
        <option
          class="jsx-1813624227"
          value="all"
        >
          All flavour
        </option>
        <option
          class="jsx-1813624227"
          value="Aniseed"
        >
          Aniseed
        </option>
        <option
          class="jsx-1813624227"
          value="Apple"
        >
          Apple
        </option>
        <option
          class="jsx-1813624227"
          value="Bitter"
        >
          Bitter
        </option>
        <option
          class="jsx-1813624227"
          value="Botanical"
        >
          Botanical
        </option>
        <option
          class="jsx-1813624227"
          value="Cherry"
        >
          Cherry
        </option>
        <option
          class="jsx-1813624227"
          value="Citrus"
        >
          Citrus
        </option>
        <option
          class="jsx-1813624227"
          value="Coconut"
        >
          Coconut
        </option>
        <option
          class="jsx-1813624227"
          value="Coffee"
        >
          Coffee
        </option>
        <option
          class="jsx-1813624227"
          value="Coke"
        >
          Coke
        </option>
        <option
          class="jsx-1813624227"
          value="Creamy"
        >
          Creamy
        </option>
        <option
          class="jsx-1813624227"
          value="Frozen"
        >
          Frozen
        </option>
        <option
          class="jsx-1813624227"
          value="Fruity"
        >
          Fruity
        </option>
        <option
          class="jsx-1813624227"
          value="Ginger"
        >
          Ginger
        </option>
        <option
          class="jsx-1813624227"
          value="Ice cream"
        >
          Ice cream
        </option>
        <option
          class="jsx-1813624227"
          value="Menthe"
        >
          Menthe
        </option>
        <option
          class="jsx-1813624227"
          value="Mint"
        >
          Mint
        </option>
        <option
          class="jsx-1813624227"
          value="Orange"
        >
          Orange
        </option>
        <option
          class="jsx-1813624227"
          value="Refreshing"
        >
          Refreshing
        </option>
        <option
          class="jsx-1813624227"
          value="Smokey"
        >
          Smokey
        </option>
        <option
          class="jsx-1813624227"
          value="Sour"
        >
          Sour
        </option>
        <option
          class="jsx-1813624227"
          value="Strawberry"
        >
          Strawberry
        </option>
        <option
          class="jsx-1813624227"
          value="Strong"
        >
          Strong
        </option>
        <option
          class="jsx-1813624227"
          value="Sweet"
        >
          Sweet
        </option>
        <option
          class="jsx-1813624227"
          value="Tea"
        >
          Tea
        </option>
        <option
          class="jsx-1813624227"
          value="Tropical"
        >
          Tropical
        </option>
      </select>
      <span
        class="jsx-1813624227 arrow"
        tabindex="0"
      />
    </div>
  </div>
</div>
`;
